### PR TITLE
Adding a new view to create a new user based on Kano World account

### PR DESCRIPTION
**Pull request by skarbat**
* The add account option will open a new dialog with login credentials
 * The dialog has 2 buttons: login to Kano World, create a new account via reboot
 * The login authenticates against Kano World, gives error message if failure

### CODE REVIEW

File: kano_greeter/newuser_view.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import os
+
+from kano.logging import logger
+from kano.gtk3.buttons import KanoButton, OrangeButton
+from kano.gtk3.heading import Heading
+from kano.gtk3.kano_dialog import KanoDialog
+
+from kano_greeter.last_user import set_last_user
+
+from kano_world.functions import login as kano_world_authenticate
```
Comment (tombettany) [inline kano_greeter/newuser_view.py]: @skarbat Do we need to depend on `kano-profile` and if so can't we just use that window for logging in for consistency.

Comment (tombettany) [inline kano_greeter/newuser_view.py]: Using that window might be a bit of a pain but we do need to at least depend on the package for these functions.

**Comment by tombettany** [comment]
Other than the comments above, I'm happy with this.

File: bin/kano-greeter-account [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/python
+
+# kano-greeter-account
+#
+# Copyright (C) 2014 Kano Computing Ltd.
```
Comment (alex5imon) [inline bin/kano-greeter-account]: This needs to be 2015

File: bin/kano-greeter-account [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+if __name__ == '__main__':
+
+    # Validate and collect command line options
+    enforce_root('You must be root to use {}'.format(sys.argv[0]))
+    if len(sys.argv) > 2:
+        username=sys.argv[1]
+        password=sys.argv[2]
+    else:
+        print 'Syntax: kano-greeter-account <username> <password>'
```
Comment (alex5imon) [inline bin/kano-greeter-account]: you might want to use the logger here instead?

File: kano_greeter/newuser_view.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+# password_view.py
+#
+# Copyright (C) 2014 Kano Computing Ltd.
```
Comment (alex5imon) [inline kano_greeter/newuser_view.py]: 2015

File: bin/kano-greeter-account (hunk 3 above)
Comment (tombettany) [inline bin/kano-greeter-account]: @alex5imon While it is useful to use the logger, the greeter is executed as the `lightdm` user which means that the things get logged to its home folder (something like `/var/lib/lightdm`) and so won't get sent to us anyway.

Comment (alex5imon) [inline bin/kano-greeter-account]: good point. Nevermind then

Comment (tombettany) [inline bin/kano-greeter-account]: Incidentally, everything that is printed in the greeter gets logged to `/var/log/lightdm/x-0.log` (it gets overwritten on each run but it does get logged)

Comment (tombettany) [inline bin/kano-greeter-account]: We might want to log it here and then modify the bug reporting to send these logs too (although that would probably require some special permissions).

Comment (skarbat) [inline bin/kano-greeter-account]: the script is already sudoed so that it has permissions to create the new account, so I will add logging here.

**Pull request merged**